pkg/apis/planetscale/v2/vitesskeyspace_types.go

```diff
@@ -294,9 +294,36 @@ type VitessKeyspaceStatus struct {
 	// ReshardingInProgress indicates if there is a VReplication workflow in progress.
 	ReshardingInProgress corev1.ConditionStatus `json:"reshardingInProgress,omitempty"`
 	// ActiveWorkflows is a list of the current VReplication workflows in progress.
-	ActiveWorkflows []string `json:"activeWorkflows,omitempty"`
+	ActiveWorkflows []WorkflowStatus `json:"activeWorkflows,omitempty"`
+	// ServingShards is a list of shards that are currently serving writes.
+	ServingShards []string `json:"servingShards,omitempty"`
```

Comment:
We're only going to fetch this info if resharding is active, right? If so, I think we should put it inside a `resharding:` struct so everything is grouped. Then we have a place to add other info like `sourceShards` and `targetShards` so we can tell which direction resharding is going.

As discussed below, I think we should also add a generic `conditions:` list like we did for VitessShardStatus, and move things like `ReshardingInProgress` there with full reason/message/lastTransitionTime. We can copy and/or refactor the code you wrote to manage those fields for VitessShardCondition.

Reply:
I've done part 2 now of what you mentioned. What do you think we should do if we end up getting back more than one active resharding workflow?